File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/HistoryEntity.ts
/**
 * Interface for: ProjectStart.Entity.Common.HistoryEntity
 */
export interface IHistoryEntity  {
    id: string;
    createdBy: string;
    dateCreated: Date;
    modifiedBy: string;
    dateModified?: Date;
    isRemoved: boolean;
    removedBy: string;
    dateRemoved?: Date;
}
/**
 * Base view model for ProjectStart.Entity.Common.HistoryEntity
 */
export class HistoryEntity  implements IHistoryEntity {
    id: string;
    createdBy: string;
    dateCreated: Date;
    modifiedBy: string;
    dateModified?: Date;
    isRemoved: boolean;
    removedBy: string;
    dateRemoved?: Date;
}

File: ProjectStart/ProjectStart.Entity/Common/HistoryEntity.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectStart.Entity.Common
{
    public class HistoryEntity
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }
        public string CreatedBy { get; set; }
        public DateTime DateCreated { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime? DateModified { get; set; }
        public bool IsRemoved { get; set; }
        public string RemovedBy { get; set; }
        public DateTime? DateRemoved { get; set; }
    }
}
